napari/_qt/qt_main_window.py

```diff
@@ -1197,3 +1179,27 @@ def close(self):
             self.qt_viewer.close()
             self._qt_window.close()
             del self._qt_window
+
+
+def _instantiate_dock_widget(wdg_cls, viewer: 'Viewer'):
```

Comment:
just moving this out of the body of `add_plugin_dock_widget` for ease of testing, and to keep that method from getting too big